### DISCUSSION TITLE
Guide for Beginner

### DISCUSSION THREAD

**Discussion by onesounds** [Flow-Launcher/Flow.Launcher#770] [Ideas]
1. Little Hotkey  Guide under the result.
At first, it was made to explain the function depending on the situation, but it seems to confuse. Only the context menu switch and containing folder shortcut are displayed. It is turned on by default and can be turned off in settings.

https://user-images.githubusercontent.com/6903107/138605029-62249ff8-afed-424c-8794-28a593957228.mp4

2. Help & First Launch Window
![Welcome_TypeB_01](https://user-images.githubusercontent.com/6903107/138605077-42774501-7806-4f0d-a775-8a507bec9872.jpg)
![Welcome_TypeB_02](https://user-images.githubusercontent.com/6903107/138605076-b7d5ee6e-7277-4f38-8754-6a0dbdd1add1.jpg)
![Welcome_TypeB_03](https://user-images.githubusercontent.com/6903107/138605075-345376e3-ee22-41a2-908a-0245bf220f24.jpg)
![Welcome_TypeB_04](https://user-images.githubusercontent.com/6903107/138605107-e8a123b1-fc75-4563-b565-414cb88cf3f3.jpg)
![Welcome_TypeB_05](https://user-images.githubusercontent.com/6903107/138605110-769dc4e8-377e-4252-9b02-273dd4f8cdfe.jpg)
![Welcome_TypeB_06](https://user-images.githubusercontent.com/6903107/138605115-2811f7aa-6af0-4ab4-8eb2-66dac8fcd3e2.jpg)
![Welcome_TypeB_07](https://user-images.githubusercontent.com/6903107/138605119-47b3177f-26ed-4eb6-ab9f-4a4c0bdb765d.jpg)
![Welcome_TypeB_08](https://user-images.githubusercontent.com/6903107/138605124-3a763db5-9e1a-478b-8fa6-af4db078f898.jpg)
![Welcome_TypeB_09](https://user-images.githubusercontent.com/6903107/138605132-2ede5197-e78f-4e0e-bdb4-ac87372af120.jpg)

It is a window with basic settings and function descriptions for first-time users.
1 to 8 are shown to the first time user. There is also content after that, but it can be finished at 8. If you press the menu, the list appears. It is viewed from 1 to 8 to those who want to check their functions or those who open the help menu. I think there are too many 1-8. I think six would be appropriate. so 1-6 is Essential information., after 6 is more.

- There should be a skip button on the first screen.
- I think it need too much image. (it need every section.) maybe I can use theme preview code or video clip.
- The theme and file preview may be excluded 1-6.
- It seems that there should be more explanations for basic functions. (Web search, System Commands etc.)
- Put a help mark in a location related to the setting window, and when you click it, the corresponding item appears in this window.
- The shortcut key basically needs to be shown in the hotkey menu of the setting.
- damn, wpf doesn't have this control. (Flipview, slideshow...)

So, Basic 1-6.
- Start (Language, Skip, about flow launcher)
- Quick Launch (Flow Starting hotkey)
- Search File, Apps, Web with keyboard
- Hotkeys (Small Guide on/off)
- Plugins
- Done

Information that needs to be added.
- Plugin Store & pm install
- File Preview
- Search Windows Control Panel
- Priority
- Action Kewords
- Everything
- Google Search
- Youtube Example (how to use)
- Spotify Example (how to install and use)
- Theme
- System Commands
- Shell
- Window Walker
- Gmail Search
- Custom Website Search (e.g intranet things)
- Custom Query Hotkey
- Query History
- Portable Mode

Anyway, do we really need this?

**Comment by jjw24** [comment]
**Regarding:** 1. Little Hotkey Guide under the result.

Love it!

> explain the function depending on the situation, but it seems to confuse

can we not do it for the function depending on the situation, what's causing the confusion? I see you got run as admin and f1 to preview as well, so it's more than just for the context menu navigation right?

**Reply by onesounds** [reply]
> **Regarding:** 1. Little Hotkey Guide under the result.

Shortcut keys need to be organized. If we are going to add the guide window below, we need to add it separately shortcut.

> I see you got run as admin and f1 to preview as well, so it's more than just for the context menu navigation right?

Yeah, I tried to tell other keys additionally depending on the situation.  actually I just check item string. if there '.exe' then display 'Run as Admin', if '.JPG' so 'F1 Preview' ...

> can we not do it for the function depending on the situation, what's causing the confusion? 

For example, the image preview tells you that f1, and in this case, ctrl+enter may feel like you can't use it.

The first one was made and stopped. The code is a little difficult for me.
Do we need the second Beginner/Help window?

**Comment by jjw24** [comment]
**Regarding:** 2.Help & First Launch Window

>Do we need the second Beginner/Help window?

The second Beginner/Help window you are referring to is the window containing help after the 6th one? The i dont think we need any more than that (obviously update more vital help within the 6). I think having 6 is enough and people wont bother to continue on for first time users because it may be too much to take in or impatience. I think the rest we can just put up a link that points to our [tips section](https://github.com/Flow-Launcher/docs/blob/main/usage-tips.md) in the docs. Having the rest of the help in the docs also allows the community to add/update easily since they are just md files.

**Reply by onesounds** [reply]
I agree that six are appropriate for the first run. Afterwards, the list will be shown when the user clicks View more separately or selects help. I think it's better to tell by program inside than the web. However, I agree that there is a management problem.

So now, I considering the form of bringing up web documents. I will use the document. https://docusaurus.io This kind of thing looks good, so I'm looking around. Language support seems convenient.

**Comment by onesounds** [comment]
https://github.com/Flow-Launcher/Flow.Launcher/pull/773

**Comment by onesounds** [comment]
https://github.com/Flow-Launcher/Flow.Launcher/pull/849
